src/System.Private.CoreLib/src/System/Diagnostics/Eventing/TraceLogging/TraceLoggingEventHandleTable.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Diagnostics.Tracing
+{
+#if FEATURE_PERFTRACING
+    /// <summary>
+    /// Per-EventSource data structure for caching EventPipe EventHandles associated with TraceLogging events.
+    /// </summary>
+    internal sealed class TraceLoggingEventHandleTable
+    {
+        private const int DefaultLength = 10;
+        private IntPtr[] m_innerTable;
+        private object m_innerTableUpdateLock = new object();
```

Comment:
Where is this used?

Reply:
Actually, upon looking at this one more time, I've added the use of it back as it is necessary.  The handle table is associated with an EventSource while NameInfo is associated with an event signature.  Thus, it is still possible even with the lock in NameInfo that two events with different signatures in the same EventSource can clobber each other if they both manipulate the handle table concurrently and one requires a re-size of the underlying IntPtr[].

Reply:
Rather than having two separate locks that both need to be taken, would it be better to just have them use the one in the handle table? Or even just lock on the handle table?

Reply:
Yes, locking on the handle table sounds like the right way to go.  We need to lock in GetOrCreateEventHandle to ensure that we don't define an additional event that isn't needed.  But, the contents of the NameInfo object doesn't change - it's essentially readonly at this point, so we don't need a lock that lives on the NameInfo.  I will give this a shot.